app/src/main/kotlin/com/wire/android/ui/calling/CallActivity.kt

```diff
@@ -0,0 +1,142 @@
+import androidx.core.view.WindowCompat
+import com.wire.android.appLogger
+import com.wire.android.notification.CallNotificationManager
+import com.wire.android.ui.AppLockActivity
+import com.wire.android.ui.LocalActivity
+import com.wire.android.ui.common.snackbar.LocalSnackbarHostState
+import com.wire.android.ui.theme.WireTheme
+import com.wire.kalium.logic.data.id.QualifiedIdMapperImpl
+import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class CallActivity : AppCompatActivity() {
+
+    @Inject
+    lateinit var callNotificationManager: CallNotificationManager
+
+    @Inject
+    lateinit var proximitySensorManager: ProximitySensorManager
+
+    private val qualifiedIdMapper = QualifiedIdMapperImpl(null)
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        callNotificationManager.hideAllNotifications()
+
+        setUpCallingFlags()
+
+        appLogger.i("$TAG Initializing proximity sensor..")
+        proximitySensorManager.initialize()
+
+        WindowCompat.setDecorFitsSystemWindows(window, false)
+
+        val conversationId = intent.extras?.getString(EXTRA_CONVERSATION_ID)
+        val screenType = intent.extras?.getString(EXTRA_SCREEN_TYPE)
+
+        setContent {
+            val snackbarHostState = remember { SnackbarHostState() }
+            CompositionLocalProvider(
+                LocalSnackbarHostState provides snackbarHostState,
+                LocalActivity provides this
+            ) {
+                WireTheme {
+                    conversationId?.let {
+                        screenType?.let { screenType ->
+                            val startDestination = CallScreenType.valueOf(screenType)
+                            CallScreen(
+                                conversationId = qualifiedIdMapper.fromStringToQualifiedID(it),
+                                startDestination = startDestination
+                            )
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    override fun onResume() {
+        super.onResume()
+        proximitySensorManager.registerListener()
+    }
+
+    override fun onPause() {
+        super.onPause()
+        proximitySensorManager.unRegisterListener()
+    }
+
+    companion object {
+        private const val TAG = "CallActivity"
+        const val EXTRA_CONVERSATION_ID = "conversation_id"
+        const val EXTRA_USER_ID = "user_id"
+        const val EXTRA_SCREEN_TYPE = "screen_type"
+    }
+}
+
+fun CallActivity.setUpCallingFlags() {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O_MR1) {
+        setShowWhenLocked(true)
+        setTurnScreenOn(true)
+    } else {
+        window.addFlags(
+            WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON
+                    or WindowManager.LayoutParams.FLAG_ALLOW_LOCK_WHILE_SCREEN_ON,
+        )
+    }
+}
+
+fun getOngoingCallIntent(
```

Comment:
why here, not in `com.wire.android.notification.PendingIntents` ??

Reply:
Because `getOngoingCallIntent()` returns an intent not a pending Intent and it's used in different screens